### DISCUSSION TITLE
Feature Request: List previously uploaded attachments

### DISCUSSION THREAD

**Discussion by TheDavidJohnson** [outline/outline#3792] [Ideas]
As a creator/editor, I want to reuse images without having to re-upload them so that I can prevent unnecessary growth of storage used by the app.

Another reason I might want to reuse an image would be because I can't recall where I grabbed the original from (was that a screenshot? did I just copy the image and paste it into Outline?) and there are times when I can't remember which doc I've used it in. (Yes, I have issues.)

One preliminary idea would be to create a `/reuse` (or similar) item in the editor that would potentially do one of the following:
* open a search UI that would allow the user to search (by filename) for a previously uploaded file
* open a list of previously uploaded images with columns for filename, date uploaded, article uploaded to, and (dangit I hate to say this  because it's a mess but if you go so far as to create the other columns, someone's going to insist on it) a thumbnai/previewl of the image. Default sort would be by most recently uploaded (in my use case it's almost always an image I uploaded very recently)

An alternative idea might be an **Attachments** item in the left-hand navigation that would allow the user to look through previously uploaded attachments. Columns could be similar to the 2nd bullet point (above), but with an additional column containing a button/link ("Use this image") which copies the markdown for the image to the clipboard so the user can go paste it in any doc they're editing at the location of their choosing.

Additional notes:
* might be another good reason to not remove file attachments #3449 
* might provide a UI for renaming attachments #3411
* could be related(?) to snippets #2030 

**Huge kudos** to you for the amazing work you're doing. Feel free to let me know if you have thoughts, questions, or if there's anything I can do to help!

**Comment by tommoor** [comment]
It's a good idea, and you've thought it out well. When you do a file upload from the block menu it should offer a selection of sources. We've also had the request to upload from a url so this would make sense in this flow too.

**Comment by ovizii** [comment]
I second this! How about an admin overview of all existing uploads and maybe a link to where they are or are not attached? So we can find uploaded files not used/linked anywhere?

**Comment by FranciscoKnebel** [comment]
Found this discussion referenced on https://github.com/outline/outline/discussions/2925 

Ran into this while migrating data from a previous platform to Outline. When testing scripts for importing, I created a ton of attachments that weren't attached to documents. Testing the API created a large number of dangling attachments that are currently stored but unreferenced by any document.

Couldn't find any clear way through the API to find these files with no references, delete the attachments and remove the files from storage.
